### PR TITLE
Feature/AKS provisioning pipeline with common variable group in Azure devops

### PR DESCRIPTION
**Pull request by prathibhapadma**
AKS provisioning pipeline with common variable group in Azure devops

### CODE REVIEW

File: scripts/pipelines/azure-devops/templates/common/kubernetes/update-variable-group.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+# Create new variable group.
+az pipelines variable-group list --organization "$1" --project "$2" --group-name "$3" --query '[0].name' -o  table | grep -q "^$3" || az pipelines variable-group create --organization "$1" --project "$2" --name "$3" --variables "$4"="$5"
+variablegroup=$(az pipelines variable-group list --organization "$1" --project "$2" --group-name "$3" --query '[0].name' -o  json)
```
[the remark this answers is not in the document]
Comment (prathibhapadma) [reply scripts/pipelines/azure-devops/templates/common/kubernetes/update-variable-group.sh:5]: @sonatype-lift ignore

**Review by albertdb — CHANGES_REQUESTED** [review]
You are deleting a lot of files that shouldn't be touched. Please fix.

File: scripts/pipelines/azure-devops/templates/aks/aks-pipeline.cfg [hunk 2]
```diff
@@ -72,9 +70,9 @@ function addPipelineVariables {
     # Add the install rancher variable.
     az pipelines variable create --name "installRancher" --pipeline-name $pipelineName --value ${installRancher}
 
-    # Add the variable to choose create or destroy AKS cluster
-    az pipelines variable create --name "operation" --pipeline-name ${pipelineName} --value ${operation}
-
+    # Add the variable to choose create or destroy AKS cluster 
+    az pipelines variable create --name "operation" --pipeline-name $pipelineName --value ${operation}
+
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/aks/aks-pipeline.cfg:75]:
```suggestion

```

File: scripts/pipelines/azure-devops/templates/aks/aks-pipeline.cfg [hunk 3]
```diff
@@ -72,9 +70,8 @@ function addPipelineVariables {
     # Add the install rancher variable.
     az pipelines variable create --name "installRancher" --pipeline-name $pipelineName --value ${installRancher}
 
-    # Add the variable to choose create or destroy AKS cluster
-    az pipelines variable create --name "operation" --pipeline-name ${pipelineName} --value ${operation}
-
+    # Add the variable to choose create or destroy AKS cluster 
+    az pipelines variable create --name "operation" --pipeline-name $pipelineName --value ${operation}
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/aks/aks-pipeline.cfg:74]:
```suggestion
    az pipelines variable create --name "operation" --pipeline-name $pipelineName --value ${operation}

```

File: scripts/pipelines/azure-devops/templates/aks/aks-provisioning.yml.template [hunk 4]
```diff
@@ -3,8 +3,8 @@ trigger:
 
 variables:
   terraformWorkingDirectory: $(System.DefaultWorkingDirectory)/.terraform/aks
-  organization: $(System.TeamFoundationCollectionUri)
-  
+  k8sVariableGroupName: 'common-k8s-variables'
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/aks/aks-provisioning.yml.template:6]:
```suggestion
  k8sVariableGroupName: 'deployment-env-variables'
```

**Review by albertdb — APPROVED** [review]

**Review by ultymatom — APPROVED** [review]

**Review by albertdb — APPROVED** [review]

**Pull request merged**